lib/orogen/gen/tasks.rb

```diff
@@ -491,6 +491,21 @@ def generate
                         returns("int").
                         base_body("    #ifdef HAS_GETTID\nreturn syscall(SYS_gettid);\n#else\nreturn 0;\n#endif").
                         doc("returns the PID for this task")
+
```

Comment:
This whole block is again (simliar to the discussion of the transformer) model knowledge and should not part of the TaskContext itself